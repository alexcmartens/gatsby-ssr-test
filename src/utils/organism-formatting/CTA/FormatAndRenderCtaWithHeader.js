import React, { useState } from "react"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import { formatCTAButtons } from "../../../components/CTAButtons/CTAButtonsConnector"
import { Link } from "gatsby"
import formatVideo from "../formatVideo"
import { CtaWithHeader } from "terra-one"
import { useContext } from "react"
import { LayoutContext } from "../../../components/Layout/Layout"

import { formatPageLinks } from "../formatPageLinks"
import ColorSchemeConnector from "../../../components/ColorSchemeConnector/ColorSchemeConnector"
import formatTheme from "../formatTheme"
const RenderCTA = ({
  data,
  paddingClass,
  overrideTheme,
  locale,
  buttons,
  variant,
  override,
  setOverride,
  products,
  siteIdentifier,
  theme,
}) => {
  return (
    <CtaWithHeader
      variant={overrideTheme || variant}
      theme={theme}
      alignment={data?.alignment}
      eyebrow={data?.eyebrow}
      headline={data?.headline}
      subhead={
        data?.longSubhead?.childMarkdownRemark?.html?.substring(0, 400) ||
        data?.subhead
      }
      content={data?.body?.childMarkdownRemark?.html || data?.body?.body}
      buttons={buttons}
      textLinks={data?.displayButtonsAsLinks}
      Link={Link}
      paddingClass={paddingClass}
      attributes={
        data?.variant === "dark" &&
        !overrideTheme && {
          theme: {
            backgroundColor: "#0063A3",
          },
        }
      }
    />
  )
}

const FormatAndRenderCtaWithHeader = ({
  data,
  paddingClass,
  overrideTheme,
  nested,
  locale = "en",
  disablePadding = false,
  marginTopAndBottom,
  productPage,
  currentSpaceId,
  products,
  siteIdentifier,
}) => {
  const { homepageSlug } = useContext(LayoutContext)
  const [override, setOverride] = useState(false)
  const appendClass = data?.extraClass ? data.extraClass : ""
  const buttons = formatCTAButtons(
    data?.ctaButtons,
    locale,
    override,
    setOverride,
    homepageSlug,
    currentSpaceId,
    products,
    siteIdentifier
  )
  const productPageComponent = productPage ? true : false
  const contentWidth = productPage ? true : data?.contentWidth || false
  const variant =
    data?.variant?.split(" ").join("-").toLowerCase() ||
    "transparent-light-text"

  const theme = ColorSchemeConnector(data?.colorScheme)
  const dataWithTheme = { ...data, ...theme }

  const formattedTheme = formatTheme(dataWithTheme)

  return disablePadding === true ? (
    <RenderCTA
      data={data}
      paddingClass={data?.paddingClass}
      overrideTheme={overrideTheme}
      locale={locale}
      buttons={buttons}
      variant={variant}
      override={override}
      setOverride={setOverride}
      products={products}
      siteIdentifier={siteIdentifier}
      theme={formattedTheme}
    />
  ) : (
    renderPaddingOrFullBleed(
      <RenderCTA
        data={data}
        paddingClass={paddingClass}
        overrideTheme={overrideTheme}
        locale={locale}
        buttons={buttons}
        variant={variant}
        override={override}
        setOverride={setOverride}
        products={products}
        siteIdentifier={siteIdentifier}
        theme={formattedTheme}
      />,
      contentWidth,
      true,
      true,
      nested,
      marginTopAndBottom,
      null,
      appendClass,
      productPageComponent
    )
  )
}

export default FormatAndRenderCtaWithHeader
