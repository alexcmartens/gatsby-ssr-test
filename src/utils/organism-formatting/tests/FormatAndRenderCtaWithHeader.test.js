import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderCtaWithHeader from "../CTA/FormatAndRenderCtaWithHeader"

describe("FormatAndRenderCtaWithHeader", () => {
  it("renders correctly", () => {
    const data = {
      variant: "dark",
      alignment: "left",
      eyebrow: "Eyebrow option",
      headline: "Header title",
      subhead: "Subhead can go here.",
      body: {
        childMarkdownRemark: {
          html: "<p>Anim aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat. Consectetur commodo nisi laborum voluptate. Commodo est ullamco pariatur ut nostrud pariatur. Anim aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat.</p>\n<ul>\n<li>Cillum et laboris aliquip consequat. El nu consectetur commodo nisi laborum voluptate.</li>\n<li>Cillum et laboris aliquip consequat. El nu consectetur commodo nisi laborum voluptate.</li>\n</ul>",
        },
      },
      ctaButtons: [
        {
          text: "123",
          url: "/",
        },
        {
          text: "123",
          url: "/",
        },
      ],
    }

    const tree = renderer
      .create(<FormatAndRenderCtaWithHeader data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
