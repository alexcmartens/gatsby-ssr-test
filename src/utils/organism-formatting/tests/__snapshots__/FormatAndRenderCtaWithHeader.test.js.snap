// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderCtaWithHeader renders correctly 1`] = `
<div
  alignment="left"
  buttons={
    Array [
      Object {
        "override": false,
        "setOverride": [Function],
        "text": "123",
        "video": undefined,
      },
      Object {
        "override": false,
        "setOverride": [Function],
        "text": "123",
        "video": undefined,
      },
    ]
  }
  content="<p>Anim aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat. Consectetur commodo nisi laborum voluptate. Commodo est ullamco pariatur ut nostrud pariatur. Anim aliquip aliqua aliqua id qui aute anim reprehenderit tempor velit est sint laborum. Cillum et laboris aliquip consequat.</p>
<ul>
<li>Cillum et laboris aliquip consequat. El nu consectetur commodo nisi laborum voluptate.</li>
<li>Cillum et laboris aliquip consequat. El nu consectetur commodo nisi laborum voluptate.</li>
</ul>"
  eyebrow="Eyebrow option"
  headline="Header title"
  subhead="Subhead can go here."
  variant="dark"
/>
`;
